rust-runtime/aws-smithy-types/src/date_time/mod.rs

```diff
@@ -301,6 +302,21 @@ impl From<SystemTime> for DateTime {
     }
 }
 
+impl PartialOrd for DateTime {
+    fn partial_cmp(&self, other: &Self) -> Option<Ordering> {
+        Some(self.cmp(other))
+    }
+}
+
+impl Ord for DateTime {
+    fn cmp(&self, other: &Self) -> Ordering {
+        match self.seconds.cmp(&other.seconds) {
+            Ordering::Equal => self.subsecond_nanos.cmp(&other.subsecond_nanos),
+            ordering => ordering,
```

Comment:
can you delegate to `as_nanos` instead? I'm 99% sure this is correct but the interaction of subsecond nanos negative dates is non trivial and if we use `as_nanos` it will definitely be correct